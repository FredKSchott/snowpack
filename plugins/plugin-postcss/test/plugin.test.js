const fs = require('fs');
const path = require('path');
const plugin = require('../plugin.js');

const cssPath = path.resolve(__dirname, './stubs/style.css');
const cssContent = fs.readFileSync(cssPath, 'utf-8');
const configFilePath = path.resolve(__dirname, './stubs/postcss.config.js');

jest.mock('execa');
const execa = require('execa');

describe('@snowpack/plugin-postcss', () => {
  beforeEach(() => {
    execa.mockClear();
    execaFn = jest.fn(() => {
      return {
        stdout: 'stdout',
        stderr: 'stderr',
      };
    });
    execa.mockImplementation(execaFn);
  });

  test('with no options', async () => {
    const pluginInstance = plugin({}, {});
    const transformCSSResults = await pluginInstance.transform({
      fileExt: path.extname(cssPath),
      contents: cssContent,
    });
    expect(execa.mock.calls[0]).toContain('postcss');
    expect(execa.mock.calls[0][1]).not.toEqual([`--config ${configFilePath}`]);
  });

  test('passing in a config file', async () => {
    const options = {
      config: path.resolve(configFilePath),
    };
    const pluginInstance = plugin({}, options);
    const transformCSSResults = await pluginInstance.transform({
      fileExt: path.extname(cssPath),
      contents: cssContent,
    });
    expect(execa.mock.calls[0]).toContain('postcss');
    expect(execa.mock.calls[0][1]).toEqual([`--config ${configFilePath}`]);
  });
});
