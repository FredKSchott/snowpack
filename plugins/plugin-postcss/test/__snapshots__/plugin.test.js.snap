// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@snowpack/plugin-postcss passing in a config file 1`] = `
"body {
    background: #fff;
}

#root {
    width: 480px;
    margin: auto;
    padding: 16px;
    box-sizing: border-box;
}

.content {
    color: blue;
}



/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0ZGluIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBO0lBQ0ksZ0JBQWdCO0FBQ3BCOztBQUVBO0lBQ0ksWUFBWTtJQUNaLFlBQVk7SUFDWixhQUFhO0lBQ2Isc0JBQXNCO0FBQzFCOztBQUVBO0lBQ0ksV0FBVztBQUNmIiwiZmlsZSI6InN0ZGluIiwic291cmNlc0NvbnRlbnQiOlsiYm9keSB7XG4gICAgYmFja2dyb3VuZDogI2ZmZjtcbn1cblxuI3Jvb3Qge1xuICAgIHdpZHRoOiA0ODBweDtcbiAgICBtYXJnaW46IGF1dG87XG4gICAgcGFkZGluZzogMTZweDtcbiAgICBib3gtc2l6aW5nOiBib3JkZXItYm94O1xufVxuXG4uY29udGVudCB7XG4gICAgY29sb3I6IGJsdWU7XG59XG5cblxuIl19 */"
`;

exports[`@snowpack/plugin-postcss with no options 1`] = `
"body {
    background: #fff;
}

#root {
    width: 480px;
    margin: auto;
    padding: 16px;
    box-sizing: border-box;
}

.content {
    color: blue;
}



/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInN0ZGluIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBO0lBQ0ksZ0JBQWdCO0FBQ3BCOztBQUVBO0lBQ0ksWUFBWTtJQUNaLFlBQVk7SUFDWixhQUFhO0lBQ2Isc0JBQXNCO0FBQzFCOztBQUVBO0lBQ0ksV0FBVztBQUNmIiwiZmlsZSI6InN0ZGluIiwic291cmNlc0NvbnRlbnQiOlsiYm9keSB7XG4gICAgYmFja2dyb3VuZDogI2ZmZjtcbn1cblxuI3Jvb3Qge1xuICAgIHdpZHRoOiA0ODBweDtcbiAgICBtYXJnaW46IGF1dG87XG4gICAgcGFkZGluZzogMTZweDtcbiAgICBib3gtc2l6aW5nOiBib3JkZXItYm94O1xufVxuXG4uY29udGVudCB7XG4gICAgY29sb3I6IGJsdWU7XG59XG5cblxuIl19 */"
`;
